lib/main.js

```diff
@@ -82,11 +102,9 @@ lift = function (obj, namespace, separator, transformer, _debug) {
             cast ? ('[' + cast(process.env[eKey]) + ']') : E) ;
 
         // if a cast function is there, it implies that native value is detected and we need to lookup environment
-        if (cast) {
+        if (cast && (process.env[eKey] !== undefined)) {
             // if environment has a defined variable, we replace the original value with this.
-            if (process.env[eKey] !== undefined) {
```

Comment:
This was previously in a separate `if` block because we did not want to needlessly recurse for a primitive case. Now in this logic, if `cast` is truthy, but `process.env[eKey]` is `undefined`, it will move on to the `else` block and start lifting it.

Is this intended or am I missing something?